File: ros_projects/src/server_client/py/area_client.py
#!/usr/bin/env python3
import rospy
import sys
from ros_projects.srv import RectangleArea
from ros_projects.srv import RectangleAreaRequest
from ros_projects.srv import RectangleAreaRequest

def calculate_area_client(width, height):
    rospy.wait_for_service("calculate_area")
    try:
        fnc = rospy.ServiceProxy("calculate_area", RectangleArea)
        response = fnc(width, height)
    except rospy.ServiceException as e:
        print("Could not call the service" %e)
    return response

if __name__ == "__main__":
    if len(sys.argv) == 3:
        width = float(sys.argv[1])
        height = float(sys.argv[2])
    else:
        print("Usage area_client.py width height")
        sys.exit(1)
    print("Area of  %s * %s is %s"%(width, height, calculate_area_client(width, height)))

File: ros_projects/src/server_client/cpp/area_client.cpp
#include "ros/ros.h"
#include "ros_projects/RectangleArea.h"
#include <cstdlib>

int main(int argc, char **argv)
{
  ros::init(argc, argv, "rectangle_area_client");
  if (argc != 3)
  {
    ROS_INFO("usage: rectangle_area width height");
    return 1;
  }

  ros::NodeHandle client_node;
  ros::ServiceClient client = client_node.serviceClient<ros_projects::RectangleArea>("rectangle_area");
  // Define server
  ros_projects::RectangleArea server;
  //Width param
  server.request.width = atoll(argv[1]);
  //Height param
  server.request.height = atoll(argv[2]);
  
  if (client.call(server))
  {
    ROS_INFO("Area: %ld", (long int)server.response.area);
  }
  else
  {
    ROS_ERROR("Failed to call service rectangle_area");
    return 1;
  }

  return 0;
}
